src/main/groovy/edu/oregonstate/mist/osuevents/db/LocalistDAO.groovy

```diff
@@ -0,0 +1,455 @@
+package edu.oregonstate.mist.osuevents.db
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.ObjectMapper
+import edu.oregonstate.mist.osuevents.core.Audience
+import edu.oregonstate.mist.osuevents.core.Campus
+import edu.oregonstate.mist.osuevents.core.County
+import edu.oregonstate.mist.osuevents.core.EventTopic
+import edu.oregonstate.mist.osuevents.core.EventType
+import edu.oregonstate.mist.osuevents.core.Location
+import edu.oregonstate.mist.osuevents.core.PaginatedCampuses
+import edu.oregonstate.mist.osuevents.core.PaginatedLocations
+import edu.oregonstate.mist.osuevents.core.PagniatedDepartments
+import org.apache.http.HttpResponse
+import org.apache.http.HttpStatus
+import org.apache.http.client.HttpClient
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.util.EntityUtils
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import javax.ws.rs.core.UriBuilder
+
+/**
+ * DAO layer for interacting with localist via its REST API
+ */
+class LocalistDAO {
+    HttpClient httpClient
+    URI baseURI
+
+    static String localistApiVersion = "2.2"
+
+    static String filtersEndpoint = "/events/filters"
+    static String placesEndpoint = "/places"
+    static String departmentsEndpoint = "/departments"
+    String communitiesEndpoint // set in constructor as config param, depends on organizationID
+
+    ObjectMapper objectMapper = new ObjectMapper()
+
+    private static Logger logger = LoggerFactory.getLogger(this)
+
+    LocalistDAO(HttpClient httpClient, String baseURL, String organizationID) {
+        this.httpClient = httpClient
+        this.baseURI = UriBuilder.fromUri(baseURL.toURI())
+                .path("/api/${localistApiVersion}").build()
+        this.communitiesEndpoint = "/organizations/${organizationID}/communities"
+    }
+
+    /**
+     * Get event topics from filters.
+     * @return
+     */
+    List<EventTopic> getEventTopics() {
+        Filters filters = getFilters()
+
+        filters.eventTopics.collect { EventTopic.fromFilter(it) }
+    }
+
+    /**
+     * Get single event topic by ID.
+     * @param id
+     * @return
+     */
+    EventTopic getEventTopicByID(String id) {
+        Filter filter = getFilters().eventTopics.find { it.filterID == id }
+
+        if (filter) {
+            EventTopic.fromFilter(filter)
+        } else {
+            null
```

Comment:
Maybe use the ternary operator: `filter ? EventTopic.fromFilter(filter) : null` ?